web/src/message_list.js

```diff
@@ -383,10 +383,17 @@ export class MessageList {
             return;
         }
 
+        const sub = stream_data.get_sub(stream_name);
+
+        // stream was just created in this browser window. Creator is
+        // considered to be subscribed, skip bookend in this case.
+        if (sub.first_message_id === null) {
+            return;
+        }
```

Comment:
Can you explain how you know `first_message_id === null` implies it was just created in this browser window? I am not sure if there are other potential explanations.

Reply:
In `lib/onboarding.py` I read this comment:
```
    # Technical note: Each stream created in the realm creation
    # process should have at least one message declared in this
    # function, to enforce the pseudo-invariant that every stream has
    # at least one message.
```
and we send a message to `channel events` upon stream creation. So, each stream has at least one message?

and it is null only for the period stream is created & the message in "channel events" is sent.

One edge case would be to delete each message in the stream, that doesn't seem practical. If you don't want to use a stream, just archive it. A stream with no messages and  then everyone unsubscribing it seems theoretical...

Though, I agree that using the keyword "browser window" is only correct for the client creating the stream. The updated comment would be "stream was just created. Creator is..."

Reply:
I tried other approach, where we use `if (stream_create.get_name() === sub.name)` see `stream_settings_ui.js`

Reply:
Another approach which I thought involves API changes. We'll have to include the information that "stream was just created" in the `subscription:add` event and call `clear_trailing_bookend()` if present in `mark_subscribed`.

Reply:
> One edge case would be to delete each message in the stream, that doesn't seem practical

People definitely delete all the messages in a stream. Another situation is that if you're joining a stream with protected history, you might not have access to any of the messages in that stream yet.

Reply:
I feel like we should not need API changes to handle this correctly. One possibility is that the client should be updating its view of `stream.first_message_id` when it receives the event for the new message itself and the current value is `null`?

Might be worth a conversation in the "api design" stream on this detail, because it might be something we should be documenting/advertising for other clients to handle live-update for viewing a newly created stream correctly.

Reply:
Marking it as resolved as this approach is no more needed after our [recent discussion in CZO](https://chat.zulip.org/#narrow/stream/9-issues/topic/.F0.9F.8E.AF.20.22you.20subscribed.22.20notice.20in.20new.20channel/near/1842933)